main.rb

```diff
@@ -1,149 +1,138 @@
-require './student'
-require './teacher'
-require './book'
-require './rental'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'app'
 
-def create_student(age, name)
-  print('Has Parent Permission [y/n]:')
-  permission = gets.chomp
-  if permission == 'y'
-    Student.new(age, name)
-  else
-    Student.new(age, name, parent_permission: false)
-  end
-end
+class Main
+  include App
 
-def create_teacher(age, name)
-  print('Specialization:')
-  specialization = gets.chomp
-  Teacher.new(age, specialization, name)
-end
+  def initialize
+    @main_ans = 0
+    @books = []
+    @people = []
 
-def create_person
-  print('Do you want to create a student(1) or a Teacher (2) [Input the number]:')
-  input = gets.chomp
-  print('age:')
-  age = gets.chomp
-  print('Name:')
-  name = gets.chomp
-  case input
-  when '1'
-    @persons << create_student(age, name)
-  when '2'
-    @persons << create_teacher(age, name)
-  else
-    puts('Wrong Input')
-    create_person
+    puts 'Welcome to School Library App!'
+    puts
   end
-  puts('Person Created succefully')
-end
-
-def create_book
-  print('title:')
-  title = gets.chomp
-  print('Author:')
-  author = gets.chomp
-  @books << Book.new(title, author)
-  puts('Book Created succefully')
-end
 
-def list_books
-  i = 0
-  @books.each do |book|
-    puts("#{i}) title: #{book.title}, author: #{book.author}")
-    i += 1
+  def show_options
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person ID'
+    puts '7 - Exit'
+    print INPUT_MSG
   end
-end
 
-def list_persons
-  i = 0
-  @persons.each do |person|
-    if person.is_a?(Teacher)
-      print '[Teacher] '
+  # rubocop:disable Metrics
```

Comment:
- [ ] You have **disabled linter check** on your code. Please enable **all** linter check on your code to ensure that your code is robust and quality.